src/EventStore.Core/TransactionLog/Chunks/TFChunkManager.cs

```diff
@@ -289,6 +311,8 @@ public async ValueTask AddChunk(TFChunk.TFChunk chunk, CancellationToken token)
 			_chunksLocker.Release();
 		}
 
+		_onChunkSwitched(newChunk.GetChunkInfo());
```

Comment:
This is one of the reasons why `callback?.Invoke()` is better. If the callback is **null**, no need to compute a value of `GetChunkInfo` method. It makes no sense to compute the value and pass it to no-op callback.